### PR TITLE
Cherry-pick merge commits into Cess run branch

### PR DESCRIPTION
**Pull request by tcclevenger**
Incorporates PRs from [confluence chart](https://acme-climate.atlassian.net/wiki/spaces/EAMXX/pages/3920363573/PRs+to+merge+into+branch+to+be+used+for+Cess-v2) into [simulations/cess-production](https://github.com/E3SM-Project/scream/tree/simulations/cess-production) branch.

This is one alternative where I use PR merge commit. 

An alternative to https://github.com/E3SM-Project/scream/pull/2538 (exact same changes). I'll close the PR that does not get chosen.

### CODE REVIEW

**Comment by crterai** [comment]
Thanks so much for creating these two PRs. It helps to have them both so we can compare the results of the two methods. The commit history does look a lot cleaner than 2538 and makes it easier to see what features went in.

File: cime_config/machines/config_machines.xml [hunk 1]
```diff
@@ -1036,7 +1036,7 @@
     <OS>CNL</OS>
     <COMPILERS>crayclang-scream</COMPILERS>
     <MPILIBS>mpich</MPILIBS>
```
Comment (tcclevenger) [inline cime_config/machines/config_machines.xml:1038]: This is the only file that I am not sure all the necessary changes are in correctly as it is the only place that had major DIFFs between the cess-production branch and new master changes.

At @ambrad and @oksanaguba, I see you two on the git blame. Could you take a look at the `<machine MACH="frontier-scream-gpu">` entry to make sure we have what we need (specifically around the ENV changes below).

**Review by tcclevenger — COMMENTED** [review]
@crterai Let me know if you there is anything else that needs to be done with this branch.

File: cime_config/machines/config_machines.xml (hunk 1 above)
Comment (ambrad) [reply cime_config/machines/config_machines.xml:1038]: Conrad, did your editor make all the whitespace changes? If so, I recommend disabling that auto-whitespace-fix option to limit the changes made in this file. (I can filter these out on github's view, of course, but the more important issue is to minimize the footprint of the changes.)

Once you do that, my understanding is this file is supposed to look exactly like the master branch's version.

Comment (tcclevenger) [reply cime_config/machines/config_machines.xml:1038]: OK, updated to match master and remove whitespaces.

**Comment by crterai** [comment]
We decided that we wanted the merge commits if the results are the same as merging individual commits, so merging this to `simulations/cess-production`.

**Comment by tcclevenger** [comment]
@crterai Let me know if there are anymore additions to cess-production needed.

**Comment by crterai** [comment]
Thanks, @tcclevenger. Would you add the following PRs to the `simulations/cess-production` branch? 
https://github.com/E3SM-Project/scream/pull/2539
Fix relative humidity units: https://github.com/E3SM-Project/scream/pull/2544
Scorpio1.4.2: https://github.com/E3SM-Project/scream/pull/2539

**Comment by crterai** [comment]
@tcclevenger, a few more recent changes have been added to master that we'd want on the `simulations/cess-production` branch. Would you add the following PRs?

Allow locally-empty layouts in AtmosphereInputs: https://github.com/E3SM-Project/scream/pull/2546
Fixes inconsistency between IO and vertical remap: https://github.com/E3SM-Project/scream/pull/2522
Ensure IO decomp tags don't clash: https://github.com/E3SM-Project/scream/pull/2550

I don't think they should produce conflicts, but do let me know if any arise.

**Comment by crterai** [comment]
@tcclevenger - We're ready to merge in one more PR. Would you merge the following PR on master to `simulations/cess-production`?

fix shoc energy fixer - https://github.com/E3SM-Project/scream/pull/2527

**Comment by tcclevenger** [comment]
@crterai Done.

**Comment by crterai** [comment]
So quick. Thanks @tcclevenger!

**Comment by tcclevenger** [comment]
@crterai Added https://github.com/E3SM-Project/scream/pull/2561

**Comment by crterai** [comment]
Thanks, @tcclevenger . Would you also add the the following PRs? This should be the last of the PRs. 
quick fix to ensure the proper units are added to netCDF metadata: https://github.com/E3SM-Project/scream/pull/2559
Set default macmic=6 for ne256: https://github.com/E3SM-Project/scream/pull/2555
Add runtime options to control shoc - isotropic_ts: https://github.com/E3SM-Project/scream/pull/2554?

**Comment by tcclevenger** [comment]
@crterai Done.

**Comment by crterai** [comment]
Thanks @tcclevenger!

**Pull request merged**